src/Mod/Part/Gui/SoBrepFaceSet.cpp

```diff
@@ -836,9 +833,11 @@ bool SoBrepFaceSet::overrideMaterialBinding(SoGLRenderAction *action, SelContext
             }
         }
 
-        SbBool notify = enableNotify(FALSE);
-        materialIndex.setValuesPointer(matIndex.size(),&matIndex[0]);
-        if(notify) enableNotify(notify);
+        if (materialIndex.getNum() != matIndex.size() || materialIndex.getValues(0) != &matIndex[0]) {
```

Comment:
I can confirm that this PR fixes the problem. However, there are a few minor issues:
* `materialIndex.getNum() != matIndex.size()` causes a compiler warning `comparison of integers of different signs: 'int' and 'std::vector::size_type' (aka 'unsigned long') [-Wsign-compare]`
* `&matIndex[0]` is the old way to access the C array. Since C++11 it should be `matIndex.data()`